lib/src/configuration.dart

```diff
@@ -792,3 +850,84 @@ enum GeneralSyncErrorCode {
   final int code;
   const GeneralSyncErrorCode(this.code);
 }
+
+/// Contains the details for a compensating write performed by the server.
+/// {@category Sync}
+class CompensatingWriteInfo {
+  /// The type of the object which was affected by the compensating write.
+  final String objectType;
+
+  /// The reason for the server to perform a compensating write.
+  final String reason;
+
+  /// The primary key of the object which was affected by the compensating write.
+  final RealmValue primaryKey;
+
+  const CompensatingWriteInfo(this.objectType, this.reason, this.primaryKey);
```

Comment:
Should we allow users to create instances of this class. I guess we should make this ctor private

Reply:
It is only a class containing data. I don't see a reason to disable it. It is not an Error.